cmdstanpy/utils.py

```diff
@@ -206,6 +210,7 @@ def scan_column_names(fp: TextIO, config_dict: Dict, lineno: int) -> int:
     lineno += 1
     names = line.split(',')
     config_dict['column_names'] = tuple(names)
+    config_dict['num_params'] = len(names) - 1
```

Comment:
this should be put inside a test (cf line 132) 
`if is_optimizing:`
 
the columns in the csv output for the HMC sampler include more things than just `__lp` and params.  which is why when parsing the csv file for the HMC sampler output, the num_params is determined by the number of columns in the  inverse mass matrix (I think - will recheck the code)

Reply:
(Please see my comment above)

Reply:
OK, great - this makes sense.